src/main/java/com/quirko/gui/GuiController.java

```diff
@@ -252,24 +287,58 @@ public void bindScore(IntegerProperty integerProperty) {
         scoreValue.textProperty().bind(integerProperty.asString());
     }
 
+    public void bindTarget(IntegerProperty integerProperty) {
+        targetValue.textProperty().bind(integerProperty.asString());
+    }
+
+    public void bindLevelID(IntegerProperty integerProperty) {
+        levelID.textProperty().bind(integerProperty.asString());
+    }
+
     public void gameOver() {
         timeLine.stop();
         gameOverPanel.setVisible(true);
         isGameOver.setValue(Boolean.TRUE);
+    }
 
+    public void nextGame(int matrixCOUNT) {
+        levelSuccPanel.setLeftMatrixCount(matrixCOUNT);
+        levelSuccPanel.setVisible(true);
+        timeLine.stop();
+        gamePanel.requestFocus();
+        isPause.setValue(Boolean.TRUE);
     }
 
     public void newGame(ActionEvent actionEvent) {
         timeLine.stop();
         gameOverPanel.setVisible(false);
-        eventListener.createNewGame();
+        eventListener.createNewGame(false);
         gamePanel.requestFocus();
         timeLine.play();
         isPause.setValue(Boolean.FALSE);
         isGameOver.setValue(Boolean.FALSE);
     }
 
+
     public void pauseGame(ActionEvent actionEvent) {
         gamePanel.requestFocus();
     }
+
+    public void setSpeed(int level) {
+        timeLine.setRate(timeLine.getRate() + (level * 1.0 * level) / 30);
+    }
+
+    public void downShift(int level) {

```

Comment:
downShift ne demek?

Reply:
Halen cevap vermemişsin

Reply:
Hocam kusura bakmayın, downShift methodunun üstüne kodda comment eklemiştim buraya yazmadı unutmuşum.
 //This method slows down (downshifts) the game difficulty when user goes to previous levels.